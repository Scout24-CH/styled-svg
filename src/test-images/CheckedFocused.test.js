import React from 'react'
import TestRenderer from 'react-test-renderer'
import Image from './CheckedFocused.js'

describe('CheckedFocused.svg generated styled component', () => {
  let renderer
  beforeEach(() => {
    renderer = TestRenderer.create(<Image />)
  })

  it('renders a <svg> tag without crashing', () => {
    expect(renderer.root.findAllByType('svg').length).toBe(1)
  })

  it('renders correctly according to snapshot', () => {
    expect(renderer.toJSON()).toMatchSnapshot()
  })

  it('has dimensions greater than zero', () => {
    const dimensions = Image.getDimensions()
    expect(dimensions.width).not.toBe('0')
    expect(parseInt(dimensions.width, 10)).toBeGreaterThan(0)
    expect(dimensions.height).not.toBe('0')
    expect(parseInt(dimensions.height, 10)).toBeGreaterThan(0)
  })
})
