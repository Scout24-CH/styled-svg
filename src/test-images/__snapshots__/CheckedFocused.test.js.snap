// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckedFocused.svg generated styled component renders correctly according to snapshot 1`] = `
<svg
  className="sc-bdVaJa krLNZu"
  viewBox="0 0 14 14"
>
  <defs>
    <circle
      cx="7"
      cy="7"
      id="s-c19cde862f-a"
      r="7"
    />
  </defs>
  <g
    fill="none"
    fillRule="evenodd"
  >
    <use
      fill="#FFF"
      xlinkHref="#s-c19cde862f-a"
    />
    <circle
      cx="7"
      cy="7"
      r="6.5"
      stroke="#FF7500"
    />
    <circle
      cx="7"
      cy="7"
      r="7.5"
      stroke="#FFF"
    />
    <circle
      cx="7"
      cy="7"
      fill="#FF7500"
      r="3"
    />
  </g>
</svg>
`;
